packages/config-plugins/README.md

```diff
@@ -164,3 +164,172 @@ export const withIcons: ConfigPlugin = config => {
 Be careful using `withDangerousModifier` as it is subject to change in the future.
 The order with which it gets executed is not reliable either.
 Currently dangerous mods run first before all other modifiers, this is because we use dangerous mods internally for large file system refactoring like when the package name changes.
+
+## Static Plugins
+
+You don't need to migrate away from your existing `app.json` or `app.config.json` to use config plugins (although it is recommended). Plugins can be resolved automatically using the JSON `plugins` array:
+
+```js
+{
+  name: 'My App',
+  plugins: [
+    // A config plugin!
+    'expo-splash-screen'
+  ]
+}
+```
+
+Static plugins can be added in two different formats:
+
+```js
+{
+  plugins: [
+    // Long form with properties
+    ['my-plugin', { /* Passed as the second parameter to the config plugin */ }]
+    // Short hand -- gets normalized to ['my-plugin', undefined]
+    'my-plugin',
+  ];
+}
+```
+
+> 💡 Everything in the Expo config must be able to be converted to JSON (with the exception of the `mods` field). So no async functions!
+
+### Module Resolution
+
+Static plugins can be resolved in a few different ways. Here are the different patterns for strings you could pass in the `plugins` array (the first property).
+
+> Any resolution pattern that isn't specified below is unexpected behavior, and subject to breaking changes.
+
+#### Project file
+
+`'./my-config-plugin.js'`
+
+```
+╭── app.config.js ➡️ Expo Config
+╰── my-config-plugin.js ➡️ ✅ `module.exports = (config) => config`
+```
+
+#### Node resolve
+
+`'expo-splash-screen'`
+
+```
+╭── app.config.js ➡️ Expo Config
+╰── node_modules/expo-splash-screen/ ➡️ Module installed from NPM (works with Yarn workspaces as well).
+    ├── package.json ➡️ The `main` file points to `build/index.js`
+    ╰── build/index.js ➡️  ✅ Node resolves to this module.
+```
+
+Sometimes you want your package to export React components and also support a plugin, to support this, multiple entry points are used. If a `app.plugin.js` file is present in the Node module's root folder, it'll be used instead of the package's main file.
+
+```
+╭── app.config.js ➡️ Expo Config
+╰── node_modules/expo-splash-screen/ ➡️ Module installed from NPM (works with Yarn workspaces as well).
+    ├── package.json ➡️ The `main` file will be used if `index.expo-plugin.js` doesn't exist.
+    ├── app.plugin.js ➡️ ✅ `module.exports = (config) => config` -- must export a function.
+    ╰── build/index.js ➡️ ❌ Ignored because `index.expo-plugin.js` exists. This could be used with `expo-splash-screen/build/index.js`
+```
+
+#### Project folder
+
+`'./my-config-plugin'`
+
+This is different to how Node module's work because `app.plugin.js` won't be resolved by default in a folder. You'll have to manually specify the file to use it.
```

Comment:
```suggestion
This is different to how Node modules work because `app.plugin.js` won't be resolved by default in a directory. You'll have to manually specify `./my-config-plugin/app.plugin.js` to use it, otherwise `index.js` in the directory will be used.
```